pkg/controller/installation/installation_controller.go

```diff
@@ -40,13 +40,23 @@ func Add(mgr manager.Manager, products []string) error {
 
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager, products []string) reconcile.Reconciler {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
 	restConfig := controllerruntime.GetConfigOrDie()
 	coreClient, err := kubernetes.NewForConfig(restConfig)
 	if err != nil {
 		logrus.Infof("error creating core client: %v", err)
 		return &ReconcileInstallation{client: mgr.GetClient(), scheme: mgr.GetScheme()}
 	}
-	return &ReconcileInstallation{client: mgr.GetClient(), scheme: mgr.GetScheme(), coreClient: coreClient, restConfig: restConfig, productsToInstall: products}
+	return &ReconcileInstallation{
+		client:            mgr.GetClient(),
+		scheme:            mgr.GetScheme(),
+		coreClient:        coreClient,
+		restConfig:        restConfig,
+		productsToInstall: products,
+		context:           ctx,
```

Comment:
It is not considered good practice to pass context via a struct property. The general accepted pattern is to pass it as the first argument in a function call
``` Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. ```  https://golang.org/pkg/context/

Reply:
I have to store it here, for it to be the same context across multiple reconciles. If I create it in the Reconcile function it would be a different context on each reconcile and the call to cancel wouldn't cancel the other reconciles context.

Reply:
it has to be stored at some point, right? Whatever owns the context needs to store it... 

The installation_controller is what owns this context.